RummikubGameApp/RummikubGameApp/GameLogic.cs

```diff
@@ -0,0 +1,340 @@
+                count++;
+            }
+            int handlength = 0;
+            while (hand[handlength] != null)
+                handlength++;
+            hand[handlength] = tilesleft[this.tilesIndex];
+            this.tilesIndex++;
+            this.currentPlayerIndex++;
+            return hand[handlength];
+        }
+
+
+        public Tile[] Shuffle2()
+        {
+            int index = 0;
+            Tile[] tilesleft = new Tile[106];
+            for (int i = 1; i <= 13; i++)
+            {
+                tilesleft[index++] = new Tile { Color = "Red", Value = i };
+                tilesleft[index++] = new Tile { Color = "Red", Value = i };
+                tilesleft[index++] = new Tile { Color = "Blue", Value = i };
+                tilesleft[index++] = new Tile { Color = "Blue", Value = i };
+                tilesleft[index++] = new Tile { Color = "Yellow", Value = i };
+                tilesleft[index++] = new Tile { Color = "Yellow", Value = i };
+                tilesleft[index++] = new Tile { Color = "Black", Value = i };
+                tilesleft[index++] = new Tile { Color = "Black", Value = i };
+            }
+            tilesleft[index++] = new Tile { Color = "Joker", Value = 0 };
+            tilesleft[index++] = new Tile { Color = "Joker", Value = 1 };
+
+            Random random = new Random();
+            for (int i = 0; i < tilesleft.Length; i++)
+            {
+                int j = random.Next(i, tiles.Length);
+                Tile temp = tilesleft[i];
+                tilesleft[i] = tilesleft[j];
+                tilesleft[j] = temp;
+            }
+            int handlength = 0, count = 0;
+            while (hand[handlength] != null)
+                handlength++;
+            for (int i = 0; i < tilesleft.Length; i++)
+            {
+                for (int j = 0; j < handlength; j++)
+                {
+                    if (hand[j] == tilesleft[i])
+                    {
+                        Tile temp = tilesleft[count];
+                        tilesleft[count] = tilesleft[i];
+                        tilesleft[i] = temp;
+                    }
+                }
+            }
+            return tilesleft;
+        }
+        
+        public void PlayTile(string name, int row, int col)
+        {
+
+            Tile tile = new Tile();
+            string tileColor;
+            string resultString = Regex.Match(name, @"\d+").Value;
+            int tileValue = Int32.Parse(resultString);
+            if (tileValue < 10)
+            {
+                tileColor = name.Substring(3);
+            }
+            else
+            {
+                tileColor = name.Substring(4);
+            }
+            tileColor = char.ToUpper(tileColor[0]) + tileColor.Substring(1);
+            tile.Color = tileColor;
+            tile.Value = tileValue;
+
+            deck[row, col] = tile;
+        }
+
+        public void AIPlayTile(string name, int row, int col)

```

Comment:
what is the diff between PlayTile and AIPlayTile? only the tileColor creation? why isn't this the same method?